### PR TITLE
Add utilities for converting responses and struct for API keys.

### PR DESCRIPTION
**Pull request by ghost**
PR adds new utilities as well as support for a struct called "OpenWeatherMapAPIKey" which is made specifically for API keys.

### CODE REVIEW

File: src/OpenWeatherMapApi/OpenWeatherMapResources.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+			}
+			public class KeyUtils
+            {
+				public static void DisposeKey(OpenWeatherMapAPIKey id)
+                {
+					id.KeyID = string.Empty;
+                }
+            }
+			public class PreparationEngine
+			{
```
Comment (WilliamApted) [inline src/OpenWeatherMapApi/OpenWeatherMapResources.cs:24]: Could you add to your PR the reasoning for adding this struct?

File: src/OpenWeatherMapApi/OpenWeatherMapUtil.cs [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+				public string Temp(CurrentWeatherResponse response)
+                {
+					return response.Main.Temp.Value.ToString();
+                }
+            }
+        }
+		/// <summary>
+		/// Booleans that return true or false based on certain conditions. Class C is for Celsius and F is for Fahrenheit.  
+		/// </summary>
+		public class Phrase
```
Comment (WilliamApted) [inline src/OpenWeatherMapApi/OpenWeatherMapUtil.cs:143]: Should this be part of an API library?

File: src/OpenWeatherMapApi/OpenWeatherMapUtil.cs [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+				List<double> coordlist = new List<double>();
+				/// Latitude comes first in the list.
+				coordlist.Add(response.Coord.Lat.Value);
+				/// Longitude comes second.
+				coordlist.Add(response.Coord.Lon.Value);
+				/// Finally, after the list is built, this method returns an array.
+				return coordlist.ToArray();
+            }
+        }
+		public class Tester
```
Comment (WilliamApted) [inline src/OpenWeatherMapApi/OpenWeatherMapUtil.cs:76]: What purpose does this class serve??

File: src/OpenWeatherMapApi/OpenWeatherMapUtil.cs [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+		/// Converter to convert CurrentWeatherResponse's to string[]'s.
+		/// </summary>
+		public class Converter
+		{
+			/// <summary>
+			/// Converts your weather response to a string[] with all information. View the code for the order the array is built.
+			/// </summary>
+			/// <param name="weatherresponse">Your weather response</param>
+			/// <returns>string[]</returns>
+			public string[] Convert(CurrentWeatherResponse weatherresponse)
```
Comment (WilliamApted) [inline src/OpenWeatherMapApi/OpenWeatherMapUtil.cs:23]: Wouldn't this be better implemented as a ToString or ToStringArray function in the WeatherResponse object?

File: src/OpenWeatherMapApi/OpenWeatherMapClient.cs [hunk 5]
```diff
@@ -14,12 +16,12 @@ public class OpenWeatherMapClient
 		private readonly HttpClient _client;
 
 		/// <summary>
-		/// Initializes a client to retireve data from OpenWeaterMap
+		/// Initializes a client to retrieve data from OpenWeaterMap
 		/// </summary>
-		/// <param name="apiKey">Your OpenWeatherMap Api Key</param>
-		public OpenWeatherMapClient(string apiKey, HttpClient client = null)
+		/// <param name="apiKey">Your OpenWeatherMap Api Key in the struct OpenWeatherMapAPIKey.</param>
+		public OpenWeatherMapClient(OpenWeatherMapAPIKey apiKey, HttpClient client = null)
```
Comment (WilliamApted) [inline src/OpenWeatherMapApi/OpenWeatherMapClient.cs:22]: Doesn't maintain compatibility with where its currently used. Major version incrementation or add a constructor taking the api key as a string still.

File: src/OpenWeatherMapApi/OpenWeatherMapResources.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace OpenWeatherMapApi
+{
+	public class APIKeys
+	{
+		public struct OpenWeatherMapAPIKey
+		{
+			private string KeyID { get; set; }
```
Comment (WilliamApted) [inline src/OpenWeatherMapApi/OpenWeatherMapResources.cs:11]: What convention are we sticking to here. Private variable so should probably be `_keyID`

File: src/OpenWeatherMapApi/OpenWeatherMapUtil.cs [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+				/// Longitude comes eleventh.
+				w.Add(weatherresponse.Coord.Lon.Value.ToString());
+				/// Finally, after the list is built, this method returns an array.
+				return w.ToArray();
+			}
+		}
+		/// <summary>
+		/// Finds coords from CurrentWeatherResponse's.
+		/// </summary>
+		public class CoordFinder
```
Comment (WilliamApted) [inline src/OpenWeatherMapApi/OpenWeatherMapUtil.cs:58]: Again, something probably better implemented in the weatherResponse class.

**Review by WilliamApted — CHANGES_REQUESTED** [review]
I've suggested some changes and have some questions 😃

**Comment by ghost** [comment]
Alright, I only added the struct because I think it would be better to organize API keys into their own structs instead of strings.
I will move all the requested methods into the WeatherResponse class.
I will also add the API key as an optional parameter.
Another thing would be to modify the classes to your requests.

**Comment by ghost** [comment]
Closing the PR because fixing it was a bit too hard.

**Pull request closed without merging**